WordPress/UITestsFoundation/Screens/Editor/BlockEditorScreen.swift

```diff
@@ -142,23 +148,43 @@ public class BlockEditorScreen: BaseScreen {
         if FancyAlertComponent.isLoaded() {
             try FancyAlertComponent().acceptAlert()
         } else {
+            let publishNowButton = app.buttons["Publish Now"]
             publishNowButton.tap()
         }
     }
 
     static func isLoaded() -> Bool {
-        return XCUIApplication().navigationBars["Gutenberg Editor Navigation Bar"].buttons["Close"].exists
+        (try? BlockEditorScreen().isLoaded) ?? false
     }
 
     @discardableResult
-    public func openBlockPicker() -> BlockEditorScreen {
+    public func openBlockPicker() throws -> BlockEditorScreen {
         addBlockButton.tap()
-        return BlockEditorScreen()
+        return try BlockEditorScreen()
     }
 
     @discardableResult
-    public func closeBlockPicker() -> BlockEditorScreen {
+    public func closeBlockPicker() throws -> BlockEditorScreen {
         editorCloseButton.coordinate(withNormalizedOffset: CGVector(dx: 0, dy: 0)).tap()
-        return BlockEditorScreen()
+        return try BlockEditorScreen()
+    }
+}
+
+// TODO: Move this to XCUITestHelpers or ScreenObject
+extension XCUIElement {
+
+    func waitFor(
+        predicateString: String,
+        timeout: TimeInterval = 10
+    ) -> XCTWaiter.Result {
+        XCTWaiter.wait(
+            for: [
+                XCTNSPredicateExpectation(
+                    predicate: NSPredicate(format: predicateString),
+                    object: self
+                )
+            ],
+            timeout: timeout
+        )
     }
```

Comment:
Extracting this is something I'll do in a followup PR (which might occur later on, or the next time I'll touch this file)